mount/mount_windows.go

```diff
@@ -10,10 +15,61 @@ func (m *Mount) Mount(target string) error {
 	return ErrNotImplementOnWindows
 }
 
+// All mounts all the provided mounts to the provided target
+func All(mounts []Mount, target string) error {
```

Comment:
Afaics this function doesn't actually mount anything and the actual mount is in https://github.com/containerd/containerd/pull/1613/files#diff-c3cba80ac2b534161f3d1259da272c21R80 and doesn't have really much to do with snapshots as all the inputs come from a spec. This makes the API very inconsistent for anything that calls this function directly instead of just creating a container what internally calls here and atm there isn't a public method for mounting a snapshot.

Reply:
This does do the mount to host (`PrepareLayer` mounts the layers), but on Windows, you cannot specify a target location. Instead, the layers are exposed as a VHD mounted on the host. The linked code just retrieves the already mounted path from HCS.

I agree that this function does not do what it says on the box here on Windows. We'll need to work out a way to expose what this really does via a different function, or otherwise make it do what it says. Having `target` as a parameter to the mount does not really make sense on Windows layer mounting.